pipeline/pipeline_tasks/api_calls/civics_control.py

```diff
@@ -0,0 +1,244 @@
+"""Control functions for Google Civics external API
+
+Implements methods as required by the
+pipeline/pipeline_api_controller.py script.
+
+Has all knowledge of the Google Civics API security requirements,
+limitations and how the data is stored.  Depends on the
+sba_sfdo_api_calls table being in place with one record for each
+record in the sba_sfdo table.
+
+"""
+import os
+import re
+import sys
+
+import time
+import datetime
+
+import pandas as pd
+import requests
+
+
+from utilities.db_manager import DBManager
+
+
+# Check that all required envars are set. Returns true if envars are set, false otherwise.
+# NOTE: Does not validate that the envar value actually allows API access.
+def check_credentials(args):
+    try:
+        if args.civics_key is None and os.environ['GOOGLEAPI'] is None:
+            return False
+    except:
+        return False
+    return True
+
+
+# Set the parameters for Google Civics. The maximum number of queries per day is 25,000.
+# We will store the data no more than 21 days.
+# The user can specify values on the command line, but only values smaller than the max are honored.
+# Returns params in a dictionary.
+def get_params(max_records, older_than):
+    params = { 'max_records': 25000, 'max_days_to_store': 21 }
+    if max_records > 0:
+        params['max_records'] = min(params['max_records'], max_records)
+    if older_than > 0:
+        params['max_days_to_store'] = min(params['max_days_to_store'], older_than)
+    return params
+
+
+# This actually gets the records to update, calls the API function and writes back to the database.
+# Returns the number of records updated, or None if a serious error occurred.
+# Can return 0 if nothing found to update.
+def update_records(args, api_params, db_params):
+    # Create a DB manager object and a pandas dataframe with just the set of records to be updated.
```

Comment:
very nit: usually comments to describe functions like this are formatted as a docstring denoted by triple quotation marks. For example,

"""
Create a DB manager object and a pandas dataframe with just the set of records to be updated.

Then call the Google Civics API for each entry in the dataframe and update if it works.
"""

Reply:
I can fix those up when I make other changes.